frontend/components/sign-message-dialog/sign-message-dialog.tsx

```diff
@@ -0,0 +1,37 @@
+import { Dialog } from '@vegaprotocol/ui-toolkit'
+import { useState } from 'react'
+import { useJsonRpcClient } from '../../contexts/json-rpc/json-rpc-context'
+import { RpcMethods } from '../../lib/client-rpc-methods'
+import { SignedMessage } from './signed-message'
+import { SignMessage } from './sign-message'
+
+export interface SignMessageDialogProps {
+  publicKey: string | null
+  onClose: () => void
+  open: boolean
+}
+
+export const SignMessageDialog = ({ publicKey, onClose, open }: SignMessageDialogProps) => {
+  const { request } = useJsonRpcClient()
+  const [signedMessage, setSignedMessage] = useState(null)
+  const [loading, setLoading] = useState(false)
+  const signMessage = async ({ message }: { message: string }) => {
+    setLoading(true)
+    const { signature } = await request(RpcMethods.SignMessage, { publicKey, message })
```

Comment:
Can this throw? Do you need a try/catch?

Reply:
Nope, this is handled by the request function. If there is an error a generic error modal is shown and the app is kept in a working state. There are no reasons this should throw, so hence no bespoke error handling